### DISCUSSION TITLE
Feature Request: On Demand Feature

### DISCUSSION THREAD

**Discussion by ChrisRichner** [wiresock/WireSockUI#8] [Ideas]
Hi there

I really like the the IOS Wireguard App On Demand Feature. 

Do you have any plans to add such a feature / behaviour to the WireSockUI?

Thanks

**Comment by wiresock** [comment]
Thank you for expressing interest in our WireSockUI app and for suggesting the integration of an **On Demand** feature. We truly value your feedback and are committed to enhancing our software based on user needs.

Your suggestion has certainly piqued our curiosity, and we're eager to learn more about your vision for this feature. Could you please provide some additional details? For instance:

- Do you envision the On Demand functionality to be limited to WiFi connections only, or would you like it to support a wider range of connection types?
- Would you prefer the feature to include extra parameters in the Wireguard configuration, such as AllowedSSID/DisallowedSSID?
- Should the On Demand feature be designed to support both WireSockUI and WireSock VPN Client in service mode?

We eagerly await your input and look forward to exploring the possibilities of incorporating this feature.

**Reply by ChrisRichner** [reply]
You're welcome!

I can elaborate a bit about the scenario that I currently use on a daily basis:

- At home (WIFI) my Wireguard VPN is off
- When I'm away on mobile connection or for any other WIFI the 'On Demand' feature kicks in and activates the one VPN configuration I have.

For the first iteration I would suggest to build the 'On Demand' feature exactly as it is today in the IOS Wireguard App. This would enable a lot of use cases like mine on a Windows Device 👍 yay 

but wait, there is even more 😀

- The 'On Demand' feature could simply look for a VPN config file which matches the WIFI SSID or Mobile Carrier Name. For example when the WIFI SSID is `sky.net` the correspondent config should be named `sky.net` as well. In case there is no such config it could fallback to a `default` file for instance.

Hope this helps to clarify my request.
